### ISSUE TITLE
Search is broken

### ISSUE THREAD

**Issue by VishDroid-dev** [shuding/nextra#2077]
**Subject:** Urgent Issue with Nextra Doc Theme - Search Function Hangs and Shows Snap Error

I hope this message finds you well. I am writing to inform you about a critical issue I am facing with the Nextra Doc theme, specifically regarding the search functionality.

Initially, after installing the theme, the search feature was working fine. However, as the content within the documentation grew, I started experiencing a significant problem. The search feature now hangs indefinitely and displays a Snap error, making it impossible to use.

To provide more clarity, I have included a screenshot of the error:

![Snap Error Screenshot](https://github.com/shuding/nextra/assets/64530007/be95100f-87c4-4d52-bb2b-3ecd77235547)

In my attempt to troubleshoot the issue, I reviewed the `theme.config.tsx` file, which contains the configuration settings for the Nextra Doc theme. Below is the relevant code snippet from the file:

```js
import React from 'react'
import { DocsThemeConfig } from 'nextra-theme-docs'

const config: DocsThemeConfig = {
  logo: <span>5P34R writeups</span>,
  project: {
    link: 'https://github.com/VishDroid-dev',
  },
  docsRepositoryBase: 'https://github.com/VishDroid-dev/myblogspace/tree/main',
  footer: {
    text: 'Made by VishDroid-dev ',
  },
  useNextSeoProps() {
    return {
      titleTemplate: '%s - 5P34R'
    }
  },
  search: {
    placeholder: 'Search Writeups',
  }
}
```

Despite adjusting various parameters within the configuration, I have been unable to resolve the issue. Therefore, I kindly request your assistance in identifying and rectifying the problem with the search functionality in the Nextra Doc theme.

Given the urgency of this matter and its impact on the usability of my documentation, I would greatly appreciate any prompt attention and support you can provide. If you require any additional information or access to my project, please do not hesitate to reach out to me.

Thank you for your immediate attention to this issue. I look forward to your response.

Best regards,  
VishDroid-dev

**Comment by VishDroid-dev** [comment]
https://0xspear.vercel.app/
Here is the link to verify yourself

**Comment by guilherssousa** [comment]
## TL;DR

Add `flexsearch.codeblocks: false` to your next.config.js, [example here](https://github.com/guilherssousa/swr-site/blob/b65349bcabb83d1ba5e89c9b6b24dd6bfd0fae8a/next.config.js#L5-L7)

## Explanation

Hi! I cloned your repo and made some tests to search for the bug and I found some interesting things:

When you click in the search bar, it starts to load content indexes using flexsearch, which allows fast text-based search on nextra. But what happens is: On `IS IT AN RSA???` page, you have some really long strings like those from `cipher.txt`, in which in just 4 lines you have 5.5K+ characters. These strings are passed into flexsearch indexing, and, internally, it performs some loops that eventually consumes all available memory, causing the website to display the out of memory error.

You probably have noticed this doesn't happen on Development environment, because dev only loads indexes for pages you already visited (that are in the .next/statics/chunk), so it won't load those heavy pages. But, during the build process, it generates this chunks for every page available on your project, thus breaking your page when you focus on Search field.

Note that this is not your fault, but maybe flexsearch isn't designed for indexing those large strings!

The fastest solution to this problem is probably moving these samples to services like Pastebin or GitHub Gists and then link them on your document. Using gist embeds you can still display the samples in your code without breaking search!

As for the maintainers, a good solution to this problem would be review the flexsearch.codeblocks configuration, it looks like it's not being respected all the times during search and still breaking the environment sometimes.

**Comment by B2o5T** [comment]
please provide a reproduction

**Comment by B2o5T** [comment]
@VishDroid-dev In your case it happened due to indexing the largest words/numbers
![image](https://github.com/shuding/nextra/assets/7361780/07f7839b-3186-4d76-8c1e-b391c2a3c1ba)

length of some of them was more than 2000 chars 😅
![image](https://github.com/shuding/nextra/assets/7361780/15e38d93-6895-4651-abe7-92505616e6de)

will push a fix!

**Comment by B2o5T** [comment]
@guilherssousa amazing deep dive into this problem 😍🎉

> As for the maintainers, a good solution to this problem would be review the flexsearch.codeblocks configuration, it looks like it's not being respected all the times during search and still breaking the environment sometimes.

I guess we should just strip out long words/number before saving it https://github.com/shuding/nextra/blob/7c8c4989021cb556a2f2f9e72b814efa311d7c2b/packages/nextra/src/mdx-plugins/remark-structurize.ts#L30-L33

**Comment by guilherssousa** [comment]
> @guilherssousa amazing deep dive into this problem 😍🎉

Thanks 🥰! The 50 character split sounds really good! Still not 100% familiar with the codebase
